test/services/katello/pulp/repository/yum_test.rb

```diff
@@ -24,7 +24,7 @@ def setup
         end
 
         def delete_repo(repo)
-          ::ForemanTasks.sync_task(::Actions::Pulp::Repository::Destroy, :pulp_id => repo.pulp_id) rescue ''
+          repo.backend_service(SmartProxy.pulp_master).delete
```

Comment:
@jlsherrill continuing from our discussion on IRC, I had to re-record the tests calling this instead of the task to get the re-record of the tests working. We talk directly to pulp in `create_repo` too so I figure its ok, but let me know if you think there is an issue doing things this way.

Reply:
delete repo is an async task, so a race condtion on re-record could cause it to fail. I'd leave it as a sync_task

Reply:
@jlsherrill but thats the thing, it wasn't working as a 'sync_task', I would even check pulp after it had run and succeeded, and the repo was still there. Let me see if I can repro on master

Reply:
On master:

Running with `ktest-live`
```
[vagrant@carrot foreman{develop}]$ bundle exec ./ktest-live ../katello/test/services/katello/pulp/repository/yum_test.rb
/home/vagrant/foreman/lib/foreman.rb:8: warning: already initialized constant Foreman::UUID_REGEXP
/home/vagrant/foreman/lib/foreman.rb:8: warning: previous definition of UUID_REGEXP was here
-- execute("SET CONSTRAINTS ALL DEFERRED;")
   -> 0.0006s
Run options: --seed 54377

# Running:

..E

Error:
Katello::Service::Repository::YumVcrTest#test_create_custom:
RestClient::Conflict: 409 Conflict
    /home/vagrant/katello/app/services/katello/pulp/repository.rb:89:in `create'
    /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:136:in `test_create_custom'
    /home/vagrant/katello/test/support/vcr.rb:24:in `block in run'
    /home/vagrant/katello/test/support/vcr.rb:23:in `run'


bin/rails test /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:125

E

Error:
Katello::Service::Repository::YumVcrCopyTest#test_copy_no_filters:
RestClient::Conflict: 409 Conflict
    /home/vagrant/katello/app/services/katello/pulp/repository.rb:89:in `create'
    /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:35:in `create_repo'
    /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:170:in `setup'
    /home/vagrant/katello/test/support/vcr.rb:24:in `block in run'
    /home/vagrant/katello/test/support/vcr.rb:23:in `run'


bin/rails test /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:180

E

Error:
Katello::Service::Repository::YumVcrCopyTest#test_errata_filter:
RestClient::Conflict: 409 Conflict
    /home/vagrant/katello/app/services/katello/pulp/repository.rb:89:in `create'
    /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:35:in `create_repo'
    /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:170:in `setup'
    /home/vagrant/katello/test/support/vcr.rb:24:in `block in run'
    /home/vagrant/katello/test/support/vcr.rb:23:in `run'


bin/rails test /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:194

E

Error:
Katello::Service::Repository::YumVcrCopyTest#test_copy_rpm_filenames:
RestClient::Conflict: 409 Conflict
    /home/vagrant/katello/app/services/katello/pulp/repository.rb:89:in `create'
    /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:35:in `create_repo'
    /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:170:in `setup'
    /home/vagrant/katello/test/support/vcr.rb:24:in `block in run'
    /home/vagrant/katello/test/support/vcr.rb:23:in `run'


bin/rails test /home/vagrant/katello/test/services/katello/pulp/repository/yum_test.rb:186

.....

Finished in 16.886943s, 0.6514 runs/s, 1.7173 assertions/s.
11 runs, 29 assertions, 0 failures, 4 errors, 0 skips
Coverage report Rcov style generated for Unit Tests to /home/vagrant/foreman/coverage/rcov
Coverage report generated for Unit Tests to /home/vagrant/foreman/coverage. 0.0 / 0.0 LOC (100.0%) covered.
```

Repo isn't being deleted in pulp
```
Mar 12 16:36:57 carrot.jomitsch.example.com pulp[2843]: celery.app.trace:INFO: [76214ed1] Task pulp.server.async.tasks._release_resource[76214ed1-0f62-447a-a115-3a954cac7471] succeeded in 0.00607972500438s: None
Mar 12 16:37:00 carrot.jomitsch.example.com pulp[26779]: pulp.server.webservices.middleware.exception:INFO: Duplicate resource: Fedora_17
Mar 12 16:37:02 carrot.jomitsch.example.com pulp[26779]: pulp.server.webservices.middleware.exception:INFO: Duplicate resource: Fedora_17
Mar 12 16:37:04 carrot.jomitsch.example.com pulp[26779]: pulp.server.webservices.middleware.exception:INFO: Duplicate resource: Fedora_17
Mar 12 16:37:05 carrot.jomitsch.example.com pulp[26779]: pulp.server.webservices.middleware.exception:INFO: Duplicate resource: Fedora_17
```

Reply:
Deleting a repo through the UI seems to work fine, but I was able to repro calling just the task on the rails console too:

repo exists in pulp
```
[vagrant@carrot katello{master}]$ pulp-admin -u admin -p admin repo list | grep zoo3
Display Name:        zoo3
```

delete from pulp in command line
```
irb(main):013:0> Katello::Repository.find(474)
=> #<Katello::Repository id: 474, pulp_id: "74bc51d3-daaf-402b-b98b-98406ca5f588", library_instance_id: nil, content_view_version_id: 8, relative_path: "sync_status_page/Library/custom/custom/zoo3", environment_id: 10, saved_checksum_type: nil, distribution_version: nil, distribution_arch: nil, distribution_bootable: nil, distribution_family: nil, distribution_variant: nil, distribution_uuid: nil, container_repository_name: nil, root_id: 67>
irb(main):014:0> Katello::Repository.find(474).pulp_id
=> "74bc51d3-daaf-402b-b98b-98406ca5f588"
irb(main):015:0> task = ::ForemanTasks.sync_task(::Actions::Pulp::Repository::Destroy, pulp_id: "74bc51d3-daaf-402b-b98b-98406ca5f588")
2019-03-12T17:04:54 [D|app|] Current user set to foreman_admin (admin)
2019-03-12T17:04:54 [D|app|] Current user set to foreman_console_admin (admin)
=> #<ForemanTasks::Task::DynflowTask id: "c6759b19-2682-45d9-99a6-25db7f29cf9c", type: "ForemanTasks::Task::DynflowTask", label: "Actions::Pulp::Repository::Destroy", started_at: "2019-03-12 17:04:54", ended_at: "2019-03-12 17:04:54", state: "stopped", result: "success", external_id: "54848503-ce68-4be6-b7aa-b5cb28bcdb33", parent_task_id: nil, start_at: "2019-03-12 17:04:54", start_before: nil, action: "Destroy">
```

repo is still there

```
[vagrant@carrot katello{master}]$ pulp-admin -u admin -p admin repo list | grep zoo3
Display Name:        zoo3
```

Reply:
I think I got it - the action doesn't take a pulp_id, rather a repository_id

Reply:
that was it! Pushed up a new commit with the change